design/dedicated-osd-pod.md

```diff
@@ -0,0 +1,66 @@
+
+- OSD Provisioner. A process that is given device or directory paths upon start and make backend specific storage types. For now, the process prepare Ceph OSDs. Currently `rook osd` command implements this functionality.
+
+### Create new OSDs
+| Sequence |Rook Operator | OSD Discovery  | OSD Provisioner   | Ceph OSD Deployment | 
+|---|---|---|---|---|
+| 1  | Read devices from cluster CRD and create an OSD Discovery deployment on each storage node  |   |   |
+| 2  | Watch Configmap  | Discover OSD devices that fit those defined in Cluster CRD and update Configmap with the device paths| | |
+| 3  | Detect Configmap change, parse Configmap, extract device paths, and create an OSD Provisioner deployment for each device  |Detect device additions or removals | | |
+| 4  | Watch Configmap |Detect device additions or removals | Prepare OSDs, Persist OSD ID, datapath, and node info in a per node Configmap | |
+| 5  | Detect Configmap change, parse Configmap, extract OSD info, construct OSD Pod command and arg | Detect device additions or removals | | |
+| 6  | Create one deployment per OSD |Detect device additions or removals | |
+| 7  | |Detect device additions or removals | | Start `ceph-osd` Daemon one Pod per device |
+
+This change addresses the above limitations in the follow ways:
+- High reliability. Each `ceph-osd` daemon runs its own Pod, their restart and upgrade are by Kubernetes controllers. Upgrading OSD Provisioner Pod no longer restarts `ceph-osd` daemons.
+- More efficient resource requests. Once OSD Discovery detects all OSD devices, Rook Operator is informed of the topology and assigns appropriate resources to each Ceph OSD deployment.
+
+### Discussions
+
+It is expected OSD Discovery process will be merged into Rook Operator once local PVs are supported in Rook Cluster CRD. Rook Operator can infer the device topoloyg from local PV Configmaps. 
+
+Alternatively, since `rook agent` is currently running as a DaemonSet on all nodes, it is conceivable to make `rook agent` to poll devices and update OSD orchestration Configmap. This approach, however, needs to give `rook agent` the privilege to modify Configmaps.
+
+## Impact
+
+- Security. OSD Provisioner Pod still uses the same RBAC rules but Ceph OSD Pod don't need to access Kubernetes resources and thus don't need any RBAC rules.
+
+- Rook Operator. Rook Operator will watch Configmaps created by OSD, reconcile Ceph OSD Daemon deployments with the latest information in Configmaps and CRD, and creates Ceph OSD deployments.
+
+- OSD Provisioner. OSD Provisioner becomes a batch job, it no longer exec Ceph OSD daemon. 
+
+- Ceph OSD Daemon. `ceph-osd` is no longer exec'ed by OSD Provisioner, it becomes the Pod entrypoint. 
```

Comment:
who creates the ceph.conf file for this ceph-osd pod entrypoint?  does the provisioner pod create the conf file as an artifact that is also consumable by the separate daemon pod?

Reply:
yes, I would expect the provisioning step would persist config for the device that is then consumed by the daemon